### ISSUE TITLE
Manifest unmarshal fails with service type

### ISSUE THREAD

**Issue by Gourab1998** [cloudfoundry-community/go-cfclient#392]
In Manifest schema, for services we have `type AppManifestService struct` array pointer as `Services`.
That works for `push operation` but when we try to unmarshal the response from generate function it fails with:

`
yaml: unmarshal errors:
  line 7: cannot unmarshal !!str xsuaa-tf into operation.AppManifestService
`  
Reason being cf controller changing services spec from object array to string array.

Example:-
cf push 

```yaml
applications:
- name: http-bin
  docker:
    image: kennethreitz/httpbin
  services:
  - name: xsuaa-tf
  processes:
  - type: web
    disk_quota: 1024M
    health-check-type: http
    health-check-http-endpoint: /get
    instances: 1
    memory: 256M
    readiness-health-check-type: http
    readiness-health-check-http-endpoint: /get
```

export GUID=$(cf app http-bin --guid)
cf curl "/v3/apps/$GUID/manifest"

```yaml
---
applications:
- name: http-bin
  docker:
    image: kennethreitz/httpbin
  services:
  - xsuaa-tf
  processes:
  - type: web
    instances: 1
    memory: 256M
    disk_quota: 1024M
    log-rate-limit-per-second: -1
    health-check-type: http
    health-check-http-endpoint: "/get"
    readiness-health-check-type: http
    readiness-health-check-http-endpoint: "/get"
```

**Comment by sneal** [comment]
This seems like a cloud controller bug since it should be the same object representation. We could implement a custom unmarshaller, but regardless this seems inconsistent and should be fixed in the API. @Gourab1998 What do you think?

**Comment by Gourab1998** [comment]
Raised an issue in controller here :- https://github.com/cloudfoundry/cloud_controller_ng/issues/3683

**Comment by ckganesan** [comment]
@Gourab1998 This issue is unrelated to the cloud controller. I have raised PR for this issue #393

**Comment by Gourab1998** [comment]
@ckganesan It's not unrelated to  cloud controller, reason being a standard REST Api resource's representation shouldn't change
Still thanks a lot for the PR, looks much easier and cleaner than what I was trying

**Comment by ckganesan** [comment]
@Gourab1998 : Please refer to this documentation link: https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#services-block. It clearly explains the service parameters mentioned in the manifest.

**Comment by sneal** [comment]
Fixed by #393
